javascript/api.js

```diff
@@ -0,0 +1,64 @@
+// 중복된 이메일 확인 요청 함수
+export async function checkDuplicateEmail(emailValue) {
+  try {
+    const response = await fetch(
+      'https://bootcamp-api.codeit.kr/api/check-email',
+      {
+        method: 'POST',
+        headers: {
+          'Content-Type': 'application/json',
+        },
+        body: JSON.stringify({
+          email: emailValue,
+        }),
+      },
+    );
+
+    return response;
+  } catch (error) {
+    console.error('Error:', error);
+    throw new Error('이메일 중복 확인 요청 실패');
+  }
```

Comment:
try-catch 문 내에서 fetch API 호출 중에 발생하는 HTTP 400 또는 500 응답과 같은 네트워크 에러를 핸들링하려는 의도로 catch 블럭에서 Error를 던지신 것 같네요.
response.ok는 HTTP 상태 코드가 200에서 299 사이인지 확인합니다. try 블럭 내에서 response.ok 가 아닌 경우에 Error를 던져서 바로 해당 에러를 catch블럭에서 잡아서 console.error를 하도록 하는 게 성훈님 의도에 더 맞을 것 같습니다. 


```suggestion
    if (!response.ok) {
      throw new Error('이메일 중복 확인 요청 실패');
    }

    return response;
  } catch (error) {
    console.error('Error:', error);
  }
```

Reply:
그런데 에러를 던져서 어떻게 활용할지 의도가 없는 것 같은데, try catch 문을 제거하고 로직을 일단 심플하게 만드는 게 더 나을 것 같습니다.

```js
export function checkDuplicateEmail(emailValue) {
  return fetch('https://bootcamp-api.codeit.kr/api/check-email', {
    method: 'POST',
    headers: {
      'Content-Type': 'application/json',
    },
    body: JSON.stringify({
      email: emailValue,
    }),
  });
}
```

Reply:
210358a7e97054bc25c72d73cf72c2c19169a4b1

말씀하신대로 에러를 던졌지만 활용할 방법은 따로 없어서 코드만 길어지게 되었네요. 필요한때만 에러를 활용하도록 해보겠습니다😅 말씀하신부분을 이용하여 좀 더 간단하게 수정하였습니다 !